tests/test_cli_compile.py

```diff
@@ -3838,3 +3889,157 @@ def test_stdout_should_not_be_read_when_stdin_is_not_a_plain_file(
     out = runner.invoke(cli, [req_in.as_posix(), "--output-file", fifo.as_posix()])
 
     assert out.exit_code == 0, out
+
+
+@pytest.mark.parametrize("input_path_absolute", (True, False))
+@pytest.mark.parametrize(
+    "test_files_collection",
+    (
+        TestFilesCollection(
+            "relative_include",
+            {
+                "requirements2.in": "small-fake-a\n",
+                "requirements.in": "-r requirements2.in\n",
+            },
+        ),
+        TestFilesCollection(
+            "absolute_include",
+            {
+                "requirements2.in": "small-fake-a\n",
+                "requirements.in": lambda tmpdir: f"-r {(tmpdir / 'requirements2.in').as_posix()}",
+            },
+        ),
+    ),
+    ids=str,
+)
+def test_second_order_requirements_path_handling(
+    pip_conf,
+    runner,
+    tmp_path,
+    monkeypatch,
+    pip_produces_absolute_paths,
+    input_path_absolute,
+    test_files_collection,
+):
+    """
+    Given nested requirements files, the internal requirements will be written
+    in the output, and it will be absolute or relative depending only on
+    whether or not the initial path was absolute or relative.
+    """
+    test_files_collection.populate(tmp_path)
+
+    # the input path is given on the CLI as absolute or relative
+    # and this determines the expected output path as well
+    if input_path_absolute:
+        input_path = (tmp_path / "requirements.in").as_posix()
+        output_path = (tmp_path / "requirements2.in").as_posix()
+    else:
+        input_path = "requirements.in"
+        output_path = "requirements2.in"
```

Comment:
Let's try being more linear?
```suggestion
    input_dir_path = tmp_path if input_path_absolute else pathlib.Path('.')
    input_path = (input_dir_path / "requirements.in").as_posix()
    output_path = (input_dir_path / "requirements2.in").as_posix()
```

Reply:
Yeah, I had to first check in REPL if that'd work as I was worried about it maybe adding a `./` in front of the string, but it worked like a charm :)